server/internal/infprocessor/infprocessor.go

```diff
@@ -293,16 +230,108 @@ func (p *P) findLeastLoadedEngine(engineIDs []string) string {
 	var minTasks int
 	var leastLoaded string
 	for _, engineID := range engineIDs {
+		if p.unavailableEngines[engineID] {
```

Comment:
Can't this be redundant as we already remove engines from `engineRouter` and `engineRouter.GetEnginesForModel` doesn't return unavailable engines?